src/SignalR/clients/csharp/Client/test/UnitTests/HttpConnectionFactoryTests.cs

```diff
@@ -17,16 +28,93 @@ public async Task ConnectionIsDisposedIfItFailsToStartAsync()
             var testHandler = new TestHttpMessageHandler(autoNegotiate: false, handleFirstPoll: false);
             testHandler.OnRequest((req, next, ct) => Task.FromException<HttpResponseMessage>(new Exception("BOOM")));
 
-            var factory = new HttpConnectionFactory(Options.Create(new HttpConnectionOptions() {
-                Url = new Uri("http://example.com"),
-                HttpMessageHandlerFactory = _ => testHandler
-            }), NullLoggerFactory.Instance);
+            var factory = new HttpConnectionFactory(
+                Mock.Of<IHubProtocol>(p => p.TransferFormat == TransferFormat.Text),
```

Comment:
Please try to avoid Moq where possible. This should easily be replaceable by `new JsonHubProtocol()`

Reply:
I considered using `new JsonHubProtocol()` when I changed this, but decided to go with `Mock.Of<IHubProtocol>(p => p.TransferFormat == TransferFormat.Text)` because it's more self-documenting. It shows what the test is actually using the IHubProtocol for.

Reply:
I'm mildly torn betweeen my preference to avoid Moq as much as possible and the fact that I do agree with @halter73 's self-documenting argument. I guess I'd consider just making a `TestTextHubProtocol` concrete class (that throws `NotImplementedException` except in the `TransferFormat` property) but I could go either way.

Reply:
I find the mock more self-documenting than the TestTextHubProtocol for something as trivial as this, because I don't have to scroll to TestTextHubProtocol to see what it's doing.

And when TestTextHubProtocol inevitably starts getting used by other tests that started life as a copy of this one, you won't be able to see the implementation for TestTextHubProtocol in the PR. And when those tests inevitably adds more requirement, you get more methods/properties that don't throw a NotImplementedException, so you don't know what's really being called by any given test even if you can find the TestTextHubProtocol implementation.

All of this for what should be a one-liner.

Reply:
I'm over Moq

Reply:
Are you suggesting I rewrite this? If so, can I at least get a reason? Even if it's something completely irrelevant like not working in environments that don't support IL emit.

Reply:
No I don't really care, we've been gradually moving away from Moq as a team. Leave it the way it is if you want but generally I've been trying to avoid Moq as well when the situation is trivial. As we move towards support more AOT scenarios Moq won't work well (or it'll work but be really slow). Though that's a future thing, nothing to worry about yet 😄